magali/_io.py

```diff
@@ -35,26 +35,93 @@ def read_qdm_harvard(path):
         coordinates are in µm and magnetic field in nT.
     """
     contents = scipy.io.loadmat(path)
+    coordinates, data_names, bz = _set_qdm_data_grid(contents)
+    data = _create_grid_qdm_data(coordinates, data_names, bz, path)
+
+    return data
+
+
+def _set_qdm_data_grid(contents):
+    """
+    Define variables for generating a grid from QDM microscopy data.
+
+    Parameters
+    ----------
+    contents: dict
+        A dictionary containing essential parameters including spacing, Bz
+        component, and sensor sample distances.
+
+    Returns
+    -------
+    coordinates: tuple of arrays
+        Arrays with coordinates of each point in the grid. Each array must
+        contain values for a dimension in the order: easting, northing,
+        vertical, etc. All arrays must be 2d and need to
+        have the same shape. These coordinates can be generated through
+        verde.grid_coordinates.
```

Comment:
```suggestion
        Tuple of 1D arrays with coordinates of each point in the grid: 
        x, y, and z (vertical). 
```

Reply:
The arrays are actually 1D. Since this is a return value, we don't describe what they "should be" but what they "are".